### PR TITLE
Subtypesof Helper Proc

### PR DESCRIPTION
**Pull request by TheDZD**
It returns a list containing all subtypes of the object path given to it, but not the object path itself. This only exists because typesof(/obj/item/thing) - /obj/item/thing is ugly as sin, and is used often for objects that have a base type which is abstract, but have non-abstract children.

Also adds a "/" to the start of a lot of the list helper procs that did not start with one.

DNM, planning to go through and look for instances that this can replace.

### CODE REVIEW

File: code/__HELPERS/lists.dm [hunk 1]
```diff
@@ -563,16 +563,20 @@ proc/dd_sortedtextlist(list/incoming, case_sensitive = 0)
...
 
 
-proc/dd_sortedTextList(list/incoming)
+/proc/dd_sortedTextList(list/incoming)
 	var/case_sensitive = 1
 	return dd_sortedtextlist(incoming, case_sensitive)
 
+/proc/subtypesof(var/path) //Returns a list containing all subtypes of the given path, but not the given path itself.
+	if(!path || !ispath(path))
+		return
```
Comment (Krausus) [inline code/__HELPERS/lists.dm]: You do **not** want this to fail silently. `CRASH` here.

File: code/game/gamemodes/malfunction/Malf_Modules.dm [hunk 2]
```diff
@@ -393,7 +393,7 @@ rcd light flash thingy on matter drain
 	var/list/possible_modules = list()
 
 /datum/module_picker/New()
-	for(var/type in typesof(/datum/AI_Module))
+	for(var/type in subtypesof(/datum/AI_Module))
```
Comment (Fox-McCloud) [inline code/game/gamemodes/malfunction/Malf_Modules.dm]: would it be necessary for this one?

Comment (TheDZD) [inline code/game/gamemodes/malfunction/Malf_Modules.dm]: Must have made this change by mistake.

**Comment by Fox-McCloud** [comment]
Right-O, this looks good to me, then.

**Pull request merged**